libpod/container.go

```diff
@@ -833,22 +826,24 @@ func (c *Container) CleanupStorage() error {
 	return c.cleanupStorage()
 }
 
+// cleanupStorage unmounts and cleans up the container's root filesystem
 func (c *Container) cleanupStorage() error {
+	if !c.state.Mounted {
+		// Already unmounted, do nothing
+		return nil
+	}
 
-	if c.state.Mounted {
-		for _, mount := range c.config.Mounts {
-			if err := unix.Unmount(mount, unix.MNT_DETACH); err != nil {
-				if err != syscall.EINVAL {
-					logrus.Warnf("container %s failed to unmount %s : %v", c.ID(), mount, err)
-				}
+	for _, mount := range c.config.Mounts {
+		if err := unix.Unmount(mount, unix.MNT_DETACH); err != nil {
+			if err != syscall.EINVAL {
+				logrus.Warnf("container %s failed to unmount %s : %v", c.ID(), mount, err)
```

Comment:
Should there be at least a debug statement for the else here?  Otherwise will it be hard to determine why the mounts are still around when they should have unmounted?

Reply:
EINVAL means the requested mount doesn't exist, which probably means it was already unmounted elsewhere

Reply:
OK, if you think it's hip as is, I'll buy!  Thx.